### PR TITLE
Updated Estonian whois.tld.ee to the new response format

### PR DESCRIPTION
**Pull request by tanelj**
Updated Estonian whois.tld.ee to the new response format (GH-489).

Reference: https://github.com/internetee/registry/blob/master/app/views/for_models/whois.erb

### CODE REVIEW

File: lib/whois/record/parser/whois.tld.ee.rb [hunk 1]
```diff
@@ -20,9 +18,111 @@ class Parser
...
+          if (content_for_scanner =~ /^Domain:\nname:\s+(.+)\n/)
+            $1.to_s.strip.downcase
+          end
+        end
+
+        property_not_supported :domain_id
+
+        property_supported :status do
+          if content_for_scanner =~ /status:\s+(.+?)\n/
+            self.class.status_mapping[$1] || $1
```
Comment (weppos) [inline lib/whois/record/parser/whois.tld.ee.rb]: As the parser is not generic but specific for this TLD, rather than setting a class-level hash I'd rather use a `case` switch. Here's an example.
https://github.com/weppos/whois/blob/master/lib/whois/record/parser/whois.cira.ca.rb#L41-L64

File: lib/whois/record/parser/whois.tld.ee.rb [hunk 2]
```diff
@@ -20,9 +18,111 @@ class Parser
...
+        class_attribute :status_mapping
+
+        self.status_mapping = {
+          'ok (paid and in zone)' => :registered,
+          'expired' => :expired
+        }
+
+        self.scanner = Scanners::WhoisTldEe
+
+        property_supported :available? do
```
Comment (weppos) [inline lib/whois/record/parser/whois.tld.ee.rb]: Please move `available?` and `registered?` after the `domain` and `domain_id`. The property definition order should be the same of https://github.com/weppos/whois/blob/master/lib/whois/record/parser/example.rb

The same applies to `disclaimer`.

File: lib/whois/record/parser/whois.tld.ee.rb [hunk 3]
```diff
@@ -20,9 +18,111 @@ class Parser
...
+              :organization => hash['name'],
+              :url          => hash['url']
+            )
+          end
+        end
+
+        property_supported :nameservers do
+          node('Name servers') do |hash|
+            Array.wrap(hash['nserver']).map do |name|
+              Nameserver.new(:name => name.downcase)
```
Comment (weppos) [inline lib/whois/record/parser/whois.tld.ee.rb]: I'd prefer to use the new Hash syntax here:

    Nameserver.new(name: name.downcase)

File: lib/whois/record/scanners/whois.tld.ee.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+      # Scanner for WhoisTldEe-based records.
+      class WhoisTldEe < Base
+
+        self.tokenizers += [
+            :scan_available,
+            :skip_head,
+            :scan_section,
+            :skip_empty_line,
+            # :skip_end,
```
Comment (weppos) [inline lib/whois/record/scanners/whois.tld.ee.rb]: Please remove this line if not relevant.

File: spec/fixtures/responses/whois.tld.ee/ee/property_status_serverhold.expected [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+#updated_on
+  %s %CLASS{time}
+  %s %TIME{2014-11-01 20:38:55 +02:00}
+
+#expires_on
+  %s %CLASS{time}
+  %s %TIME{2015-11-01}
+
+#disclaimer
```
Comment (weppos) [inline spec/fixtures/responses/whois.tld.ee/ee/property_status_serverhold.expected]: Please move the disclaimer at the top. The order of the elements in the expectation file are relevant.

File: spec/fixtures/responses/whois.tld.ee/ee/property_status_serverhold.expected [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  %s[0].fax           == nil
+  %s[0].email         == "Not Disclosed - Visit www.internet.ee for webbased WHOIS"
+  %s[0].created_on    == nil
+  %s[0].updated_on    %TIME{2010-12-09 16:08:33 +02:00}
+
+#nameservers
+  %s %CLASS{array}
+  %s %SIZE{2}
+  %s[0] %CLASS{nameserver}
+  %s[0].name == "ns4.edicy.net"
```
Comment (weppos) [inline spec/fixtures/responses/whois.tld.ee/ee/property_status_serverhold.expected]: Please also add a test for `ipv4` and `ipv6` and ensure they are `nil`.

**Comment by weppos** [comment]
Thanks for the patch @tanelj, it looks great! I left some minor comments inline to make sure the code matches the conventions used for the other parsers.

If you don't mind to make those changes, then I'll merge it!

Thanks! :clap:

**Comment by tanelj** [comment]
Thanks for the feedback. Changes have now been made.
